File: 258-AddDigits/258-AddDigits.java
/*

258. Add Digits
Easy

Given a non-negative integer num, repeatedly add all its digits until the result has only one digit.

Example:

Input: 38
Output: 2 
Explanation: The process is like: 3 + 8 = 11, 1 + 1 = 2. 
             Since 2 has only one digit, return it.

Follow up:
Could you do it without any loop/recursion in O(1) runtime?

*/

class Solution {
    public int addDigits(int num) {
        while(num >= 10){
            int sum = 0;
            while(num != 0){
                sum += num % 10;
                num = num / 10;
            }
            num = sum;
        }
        return num;
    }
}

File: 258-AddDigits/258-AddDigits.py
"""

258. Add Digits

Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.

Example 1:

Input: num = 38
Output: 2
Explanation: The process is
38 --> 3 + 8 --> 11
11 --> 1 + 1 --> 2 
Since 2 has only one digit, return it.

Example 2:

Input: num = 0
Output: 0

Constraints:

    0 <= num <= 231 - 1

 

Follow up: Could you do it without any loop/recursion in O(1) runtime?

"""

class Solution:
    def addDigits(self, num: int) -> int:
        while num > 9:
            curr = 0
            for ch in str(num): curr += int(ch)
            num = curr
        return num
